### PR TITLE
[INLONG-5708][DataProxy] Provide single flume channel to reduce synchronized operation

### LINKED ISSUE #5708: [Feature][DataProxy] Provide single flume channel to reduce synchronized operation, and improve DataProxy performance.
### Description

Provide single flume channel to reduce synchronized operation, and improve DataProxy performance(total improve is more than 15%+).
image
image

MemoryChannel,times:1000000,transactionSize:1,duration:1363
BufferQueueChannel,times:1000000,transactionSize:1,duration:84,gap:0.9383712399119589
MemoryChannel,times:1000000,transactionSize:10,duration:2818
BufferQueueChannel,times:1000000,transactionSize:10,duration:122,gap:0.9567068843151171
MemoryChannel,times:1000000,transactionSize:100,duration:19189
BufferQueueChannel,times:1000000,transactionSize:100,duration:687,gap:0.9641982385741831

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luchunliang**
### Prepare a Pull Request
- Title: [INLONG-5708][Feature][DataProxy] Provide single flume channel to reduce synchronized operation, and improve DataProxy performance
- Fixes #5708 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by gosonzhang — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/test/java/org/apache/inlong/dataproxy/channel/TestBufferQueueChannel.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        bufferContext.put("maxBufferQueueSizeKb", "50000");
+        final BufferQueueChannel bufferChannel = new BufferQueueChannel();
+        bufferChannel.configure(bufferContext);
+        bufferChannel.start();
+        // times
+        int times = 1000000;
+        // transactionSize=1
+        int transactionSize = 1;
+        long memoryDuration = runChannel(memoryChannel, times, transactionSize);
+        System.out.println(String.format("MemoryChannel,times:%d,transactionSize:%d,duration:%d", times,
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/test/java/org/apache/inlong/dataproxy/channel/TestBufferQueueChannel.java:65]: Suggested using log framework to print the log info.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-dataproxy/dataproxy-source/src/test/java/org/apache/inlong/dataproxy/channel/TestBufferQueueChannel.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                Transaction takeTx = channel.getTransaction();
+                takeTx.begin();
+                for (int j = 0; j < transactionSize; j++) {
+                    channel.take();
+                }
+                takeTx.commit();
+                takeTx.close();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/test/java/org/apache/inlong/dataproxy/channel/TestBufferQueueChannel.java:121]: Changing to use log framework, thanks.

Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/test/java/org/apache/inlong/dataproxy/channel/TestBufferQueueChannel.java]: fixed

File: inlong-dataproxy/dataproxy-source/src/test/java/org/apache/inlong/dataproxy/channel/TestBufferQueueChannel.java (hunk 1 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/test/java/org/apache/inlong/dataproxy/channel/TestBufferQueueChannel.java]: fixed

**Pull request closed without merging**